compiler/docgen.nim

```diff
@@ -505,7 +505,7 @@ proc runAllExamples(d: PDoc) =
 
 proc quoted(a: string): string = result.addQuoted(a)
 
-proc prepareExample(d: PDoc; n: PNode): tuple[rdoccmd: string, code: string] =
+proc prepareExample(d: PDoc; n: PNode, topLevel: bool): tuple[rdoccmd: string, code: string] =
```

Comment:
```suggestion
proc prepareExample(d: PDoc; n: PNode, topLevel = false): tuple[rdoccmd: string, code: string] =
```
The common case is that the example isn't at the top level.

Reply:
it doesn't simplify the code, `prepareExample` is used only once